### DISCUSSION TITLE
BeerLambertLaw.m erroring on 4 wavelength data

### DISCUSSION THREAD

**Discussion by homavahidi** [huppertt/nirs-toolbox#25] [Q&A]
Hi there! 

I have been using nirs toolbox to analyze my fNIRS data for a while. Recently, I have been running into an error when trying to go from OD to HbO/HbR. I have traced back the error to BeerLAmbertLaw.m line 221. This line would work fine if you only have 2 wavelengths, but fails if you have 4. Can anyone advise why this might be happening? I have gone back in the recent changes to the toolbox and this line doesn’t seem to have changed. Thoughts?

**Comment by Frkle** [comment]
Hey,

Could you post the error message?

**Reply by homavahidi** [reply]
Error using nirs.modules. BeerLambertLaw/runThis (Line 221)
To assign to or create a variable in a table, the number of rows must match the height of the table.
Error in nirs.modules.AbstractModule/run (line 77)
out = obj.runThis inputs

**Reply by Frkle** [reply]
And when you used it before, without any errors, it did work with 4 wavelengths?

**Reply by homavahidi** [reply]
Last time I used the toolbox on this 4 wavelength dataset was in the summer (in June). Since then I’ve only used it on 2 wavelengths datasets. But yes, when I last used it it worked fine without issues.

**Reply by Frkle** [reply]
Okay thanks, I just wanted to make sure that I understood you correctly. I guess you do not have the version of the toolbox you used last time when analysing that data set? 
I have an older version on my computer which I used beginning of last year (and the year before) and when quickly screening both versions of the BeerLambertLaw function I would agree that there are some changes made. I have at the moment no time to find the problem in the code but if you want I can send you that version and you can either simply use that older version and we hope that it will work for your data set or (if it is a working version) you could try to find the problem by yourself :)

**Reply by homavahidi** [reply]
If you can send me the older version, that would be much appreciated! My email is hvahidi2@uwo.ca.
